lib/google/protobuf.ex

```diff
@@ -134,4 +134,39 @@ defmodule Google.Protobuf do
       nanos: rem(nanoseconds, 1_000_000_000)
     })
   end
+
+  if Code.ensure_loaded?(Duration) do
+    @doc """
+    Converts a `Google.Protobuf.Duration` struct to a `Duration` struct.
+
+    ## Examples
+
+        iex> to_duration(%Google.Protobuf.Duration{seconds: 1, nanos: 500_000_000})
+        Duration.new!(second: 1, microsecond: {500_000, 6})
+
+    """
+    @spec to_duration(Google.Protobuf.Duration.t()) :: Duration.t()
+    def to_duration(%Google.Protobuf.Duration{} = duration) do
+      Duration.new!(second: duration.seconds, microsecond: {div(duration.nanos, 1_000), 6})
+    end
+
+    @doc """
+    Converts a `Duration` struct to a `Google.Protobuf.Duration` struct.
+
+    ## Examples
+
+      iex> from_duration(Duration.new!(second: 1, microsecond: {500_000, 6}))
+      ...> %Google.Protobuf.Duration{seconds: 1, nanos: 500_000_000}
```

Comment:
I think the `...>` is not meant to be there?

```suggestion
      %Google.Protobuf.Duration{seconds: 1, nanos: 500_000_000}
```